pandas/core/sorting.py

```diff
@@ -254,7 +255,13 @@ def nargsort(items, kind='quicksort', ascending=True, na_position='last'):
             sorted_idx = np.roll(sorted_idx, cnt_null)
         return sorted_idx
 
-    items = np.asanyarray(items)
+    with warnings.catch_warnings():
+        # https://github.com/pandas-dev/pandas/issues/25439
+        # can be removed once ExtensionArrays are properly handled by nargsort
+        warnings.filterwarnings(
```

Comment:
why don't we call ``.to_numpy()`` here instead? (I think think *always* is an index), or is this a performance issue? (until we have real EA support)

Reply:
I am not opposed to this as an interim solution, just want to understand

Reply:
As far as I understand, `to_numpy` has the future behaviour (i.e. converting to object dtype), while here we want to have the current M8[ns]

Reply:
But of course, with `to_numpy(dtype='datetime64[ns]')`, you can get that.

In that case, we have to do a check here for datetimetz data; something like:

```
if is_datetimetz_dtype(items):
    items = items.to_numpy(dtype='datetime64[ns]')
else:
    items = np.asanyarray(items)
```

It's also custom code for this case, but if the above is preferable, happy to change (personally don't have a strong favor for either of them).

Reply:
yeah i think this needs to be sorted, but ok for now with your existing.

Reply:
however, shall we leave this or a follow-on open to fix this/

Reply:
I would like to expand PR https://github.com/pandas-dev/pandas/pull/25595 to include a proper fix (for 0.25.0), involving a clean-up of EA.argsort, which will be the follow-up on this. 
And apparently, I already opened an issue about the argsort API problem a few months ago: https://github.com/pandas-dev/pandas/issues/21801 (will update that issue with the new information)

Reply:
ok cool.